README.adoc

```diff
@@ -0,0 +1,131 @@
+Official https://github.com/joelittlejohn/jsonschema2pojo[jsonschema2pojo] Gradle plugin
+========================================================================================
+:toc:
+:toc-placement: preamble
+:toclevels: 1
+:showtitle:
+
+// Need some preamble to get TOC:
+{empty}
+
+== Introduction
+This plugin is aiming to take raw JSON or YAML raw files or schemas and convert to Java or Scala POJOs (Plain Old Java Object).
+
+Currently it suppots features like generating using Jackson, GSON and Mocha1 annotations and JSR305.
+
+Please note, that JSR305 is purely supported, and present only if you don't want to use a proper JSON validation for some reasons.
+
+== Adding a plugin to your build.gradle
+
+[source,gradle]
+----
+plugins {
+  id "org.jsonschema2dataclass" version "2.1.0" // Please update version as new versions are out
+}
+----
+
+Please refer https://plugins.gradle.org/plugin/org.jsonschema2dataclass[Gradle Plugins page] for further details.
+
+== Usage Example
+
+Usage for Android and for Java is basically the same. You need to declare plugin usage and configure plugin parameters if needed.
+
+To execute just a java classes generation you need to run `generateJsonSchema2DataClass` gradle task. Otherwise, `compileJava` and appropriate android tasks are already dependent to this plugin.
+
+To start using a plugin you need follow this recipe:
+
+.Minimal java example how to use the plugin
+[source,gradle]
+----
+plugins{
+    id "org.jsonschema2dataclass" version "2.1.0"
```

Comment:
What's about to use

     id "org.jsonschema2dataclass" version ${LATEST_VERSION}

instead of

     id "org.jsonschema2dataclass" version "2.1.0"

to avoid maintenance ?

Reply:
people will try :) I did few years back and this won't work for Gradle DSL. I'd put a comment about this example